### PR TITLE
Update black

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR bumps the version of the `black` pre-commit hook, and adds format changes to files for the new version standard.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/pulls.html#the-iris-check-list)

### CODE REVIEW

File: lib/iris/fileformats/_ff.py [hunk 1]
```diff
@@ -42,23 +42,140 @@
...
     ("total_prognostic_fields", (153,)),
-    ("data", (160, 161, 162,)),
+    (
+        "data",
+        (
+            160,
+            161,
+            162,
+        ),
+    ),
```
Comment (trexfeathers) [inline lib/iris/fileformats/_ff.py:178]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/fileformats/pp.py [hunk 2]
```diff
@@ -105,10 +105,37 @@
...
+    ),
+    (
+        "brsvd",
+        (
+            46,
+            47,
+            48,
+            49,
+        ),
+    ),
```
Comment (trexfeathers) [inline lib/iris/fileformats/pp.py:138]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/fileformats/pp.py [hunk 3]
```diff
@@ -163,10 +190,37 @@
...
```
Comment (trexfeathers) [inline lib/iris/fileformats/pp.py:223]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/plot.py [hunk 4]
```diff
@@ -351,7 +351,12 @@ def _check_bounds_contiguity_and_mask(coord, data, atol=None, rtol=None):
             )
 
             not_masked_at_discontiguity_along_y = np.any(
-                np.logical_and(mask_invert[:-1,], diffs_along_y)
+                np.logical_and(
+                    mask_invert[
+                        :-1,
+                    ],
+                    diffs_along_y,
+                )
```
Comment (trexfeathers) [inline lib/iris/plot.py:359]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/test_cdm.py [hunk 5]
```diff
@@ -1135,7 +1135,13 @@ def assert_is_not_lazy(self, cube):
...
 
     def test_slices(self):
-        lat_cube = next(self.cube.slices(["grid_latitude",]))
+        lat_cube = next(
+            self.cube.slices(
+                [
+                    "grid_latitude",
+                ]
+            )
+        )
```
Comment (trexfeathers) [inline lib/iris/tests/test_cdm.py:1144]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/test_nimrod.py [hunk 6]
```diff
@@ -73,7 +73,14 @@ def test_huge_field_load(self):
...
             cube = iris.load(
-                tests.get_data_path(("NIMROD", "uk2km", "cutouts", datafile,))
+                tests.get_data_path(
+                    (
+                        "NIMROD",
+                        "uk2km",
+                        "cutouts",
+                        datafile,
+                    )
+                )
```
Comment (trexfeathers) [inline lib/iris/tests/test_nimrod.py:83]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/test_nimrod.py [hunk 7]
```diff
@@ -87,7 +94,14 @@ def test_load_kwarg(self):
...
             with open(
-                tests.get_data_path(("NIMROD", "uk2km", "cutouts", datafile,)),
+                tests.get_data_path(
+                    (
+                        "NIMROD",
+                        "uk2km",
+                        "cutouts",
+                        datafile,
+                    )
+                ),
```
Comment (trexfeathers) [inline lib/iris/tests/test_nimrod.py:104]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/test_pp_cf.py [hunk 8]
```diff
@@ -88,7 +88,12 @@ def _test_file(self, name):
             fname_name = name
 
         self.assertCML(
-            cubes, self._ref_dir + ("from_pp", fname_name + ".cml",)
+            cubes,
+            self._ref_dir
+            + (
+                "from_pp",
+                fname_name + ".cml",
+            ),
```
Comment (trexfeathers) [inline lib/iris/tests/test_pp_cf.py:96]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/fileformats/pp/test_PPField.py [hunk 9]
```diff
@@ -35,7 +35,18 @@
...
+        (
+            38,
+            39,
+            40,
+            41,
+            42,
+            43,
+            44,
+        ),
+    ),
```
Comment (trexfeathers) [inline lib/iris/tests/unit/fileformats/pp/test_PPField.py:49]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/fileformats/pp/test_save.py [hunk 10]
```diff
@@ -324,10 +324,14 @@ def _get_single_time_cube(set_time_mean=False):
     cube.remove_coord("time")
     cube.remove_coord("forecast_period")
     tc = DimCoord(
-        points=[15,],
+        points=[
+            15,
+        ],
```
Comment (trexfeathers) [inline lib/iris/tests/unit/fileformats/pp/test_save.py:329]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/fileformats/um/fast_load_structured_fields/test_BasicFieldCollation.py [hunk 11]
```diff
@@ -222,7 +222,12 @@ def test(self):
             for test_tuple in test_date_tuples
         ]
         # Check all values are distinct.
-        self.assertEqual(len(test_date_ints), len(set(test_date_ints),))
+        self.assertEqual(
+            len(test_date_ints),
+            len(
+                set(test_date_ints),
+            ),
```
Comment (trexfeathers) [inline lib/iris/tests/unit/fileformats/um/fast_load_structured_fields/test_BasicFieldCollation.py:229]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/fileformats/um/optimal_array_structuring/test_optimal_array_structure.py [hunk 12]
```diff
@@ -95,7 +95,13 @@ def test_2d(self):
...
         ]
         shape, primaries, elems_and_dims = optimal_array_structure(elements)
-        self.assertEqual(shape, (2, 3,))
+        self.assertEqual(
+            shape,
+            (
+                2,
+                3,
+            ),
+        )
```
Comment (trexfeathers) [inline lib/iris/tests/unit/fileformats/um/optimal_array_structuring/test_optimal_array_structure.py:104]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/fileformats/um/optimal_array_structuring/test_optimal_array_structure.py [hunk 13]
```diff
@@ -115,7 +121,13 @@ def test_2d_with_element_values(self):
...
             elements, elements_values
         )
-        self.assertEqual(shape, (2, 3,))
+        self.assertEqual(
+            shape,
+            (
+                2,
+                3,
+            ),
+        )
```
Comment (trexfeathers) [inline lib/iris/tests/unit/fileformats/um/optimal_array_structuring/test_optimal_array_structure.py:130]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

File: lib/iris/tests/unit/util/test__slice_data_with_keys.py [hunk 14]
```diff
@@ -105,7 +105,13 @@ def test_1d_all(self):
     def test_1d_tuple(self):
         # The call makes tuples into 1-D arrays, and a trailing Ellipsis is
         # added (for the 1-D case only).
-        self.check((3,), Index[(2, 0, 1),], [(np.array([2, 0, 1]), Ellipsis)])
+        self.check(
+            (3,),
+            Index[
+                (2, 0, 1),
+            ],
```
Comment (trexfeathers) [inline lib/iris/tests/unit/util/test__slice_data_with_keys.py:112]: This formatting looks really horrible, is less readable and seems gratuitous. Is there a configuration we can use to make Black less zealous?

**Comment by bjlittle** [comment]
@trexfeathers Yeah, I tend to agree... I am slightly upset with its choices... I'll tinker under the hood and experiment a little with what controls black provides to see if it alleviates these changes.

File: lib/iris/fileformats/_ff.py (hunk 1 above)
Comment (rcomer) [reply lib/iris/fileformats/_ff.py:178]: 😱

Comment (bjlittle) [reply lib/iris/fileformats/_ff.py:178]: I know, right...

Comment (rcomer) [reply lib/iris/fileformats/_ff.py:178]: Almost wondering if it's a bug in black?  Would anyone really choose this?

Comment (trexfeathers) [reply lib/iris/fileformats/_ff.py:178]: @bjlittle will likely be able to explain all once he pushes the fixing commit. He does know what's wrong

**Comment by bjlittle** [comment]
@trexfeathers I'm closing this in favour of #3866... it just make life easier :wink:

**Pull request closed without merging**